src/web/services/log.service.ts

```diff
@@ -1,7 +1,8 @@
 import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs';
+import { QueryParams } from '../app/pages-logs/logs-page.component';
```

Comment:
A service class should not be aware of component class. But the front-end architecture right now is sketchy at best, so this can stay as is.

Reply:
Would it be better if I put this interface in the service class and the component imports the interface?

Reply:
Sounds like a better alternative to me